### ISSUE TITLE
DST Staging Review (Consider): Screen reader announcement of indeterminate state may be confusing

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#103187]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Design System Staging Information
**Component:** checkbox
**Staging Review ticket:** [Design System Staging Review: Checkbox indeterminate state](https://github.com/department-of-veterans-affairs/va.gov-team/issues/102510)

## Findings details
The details of this ticket are for consideration only. No Staging Review issue was found.
**High-priority:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

Screen reader behavior for the indeterminate state is likely going to vary a lot more than more conventional input states. Examples:
- VoiceOver on iOS announces the checkbox as "mixed"
- Narrator on Windows announces the checkbox as "indeterminate"
- NVDA on Windows announces the checkbox as "half checked"

With the visual context of the full checkbox group, I think the checkbox state is relatively easy to learn. But since this is a less common state for checkboxes and screen reader users may not have that visual context, I'm not sure that a typical screen reader user is going to know what "mixed," "indeterminate," or "half checked" are trying to communicate.

## Recommended action
I think something in the guidance for this state should comment on the potential for screen reader user confusion, and emphasize that this state should only be used when there's no other appropriate pattern.

Something else to consider: Is there a case for some kind of `aria-describedby` text to provide a extra information about the state --- eg. announcing as "checkbox, mixed, 2 of 3 options selected." But I don't know if that would be more confusing or less confusing!

## References

- **Accessibility Defect Severity:** [2: High. Should be fixed in 1-2 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-2)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for DST

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Governance help validating the issue, please comment on the ticket
- If this ticket has a `high-priority` label, please address as soon as possible so VFS teams are not impacted
- If this ticket has a `consider` label, please consider for future implementation
